src/tls.c

```diff
@@ -7178,35 +7249,77 @@ static int TLSX_KeyShare_GenOqsKey(WOLFSSL *ssl, KeyShareEntry* kse)
         return BAD_FUNC_ARG;
     }
 
-    pubKey = (byte*)XMALLOC(kem->length_public_key, ssl->heap,
-                               DYNAMIC_TYPE_PUBLIC_KEY);
-    privKey = (byte*)XMALLOC(kem->length_secret_key, ssl->heap,
-                                DYNAMIC_TYPE_PRIVATE_KEY);
-    if (pubKey == NULL || privKey == NULL) {
-        WOLFSSL_MSG("memory allocation failure");
+    ecc_kse = XMALLOC(sizeof(*ecc_kse), ssl->heap, DYNAMIC_TYPE_TLSX);
```

Comment:
You must cast this to `(KeyShareEntry*)`

Reply:
done. See next commit.